### ISSUE TITLE
Alt or Ctrl + left/right arrows behaves incorrectly

### ISSUE THREAD

**Issue by leviport** [pop-os/cosmic-term#149]
Alt or Ctrl + left/right arrows should move the cursor forward or backward 1 word. Currently, it moves the cursor forward or backward by one character and inserts unwanted characters. Ctrl + arrows inserts `;5` and alt + arrows inserts `;3`

To recreate: 
- In a terminal, type a few words separated by spaces
- Hold alt or ctrl, then tap the left arrow

Version: `0.1.0~1709152706~22.04~36477e0`

**Comment by Waridley** [comment]
Shift + arrows also doesn't select text as expected.

I'm interested in working on this, but I need a bit of guidance on where this behavior should be handled. I assume https://github.com/pop-os/cosmic-term/blob/7c5d5440e5a560238de3666d7ead6970477a7dd6/src/terminal_box.rs#L641-L654 is doing exactly what it's supposed to, and those control codes should be received and handled somewhere else? I'm just not sure where control codes are actually turned into cursor and selection modifications. Clearly somewhere the codes for arrow keys *without* modifiers is being handled and moving the cursor, but it's failing to handle modifiers. I just don't know where that is.

**Comment by ziprasidone146939277** [comment]
I am trying some Cosmic's Apps until found this. Badly have to back to gnome-terminal. Splitting terminal is a good feature.

**Comment by leb-kuchen** [comment]
A simple solutions is to remove the is_app_cursor log, as ss3 sequences do work.

**Comment by leb-kuchen** [comment]
The output for `Ctrl + ArrowRight`:
```rs
println!("{:?}\n", state.modifiers.iter().collect::<Vec<_>>());
println!("{:?}\n", mode.iter().collect::<Vec<_>>());
```

``` 
[Modifiers(CTRL)]
[TermMode(SHOW_CURSOR), TermMode(BRACKETED_PASTE), TermMode(LINE_WRAP), TermMode(ALTERNATE_SCROLL), TermMode(URGENCY_HINTS)]
```
This was implemented in #122. @snaggen  Can tell us, what `is_app_cursor` is used for.
